### ISSUE TITLE
Cutter needs testing

### ISSUE THREAD

**Issue by scottkleinman** [WheatonCS/Lexos#447]
This is a merger of issues #401 and #412. I made extensive changes to the error handling in Cutter, and it seems to work pretty well. It just needs testing.

One casualty of the changes is that we've lost the timeout function Evan added. This was a good idea, so I'm going to try to add it back tomorrow.

**Comment by scottkleinman** [comment]
Small update. I've restored Evan's timeout function and improved it. After 10 seconds you now get a warning that the job is taking a long time, and you have the option to continue or cancel.

**Comment by mleblanc321** [comment]
while (re)testing DAZ, i've found an annoying and stubborn bug:  

if Daniel is uploaded first (and i set the Default cutting values to TOKENS=450) and Azarias' Individual cutting options are set as individual options (SEGMENT=1), i can reproduce the dendrogram OK;

however, if i attempt to apply the Default Cutting options to AZ (Segment=1) and set Daniel's Individual Cutting Options as TOKENS=450, i get the warning:
     _Current cut settings will result in over 100 new segments. Please be patient if you continue._

If i enter CONTINUE, i do get the correct cuts (and dendrogram)

i'm guessing(?) that in:   `scripts_cut.js`
in the function:   `var checkForWarnings = function() {`

the `needsWarning` boolean is set to true incorrectly;  

just a guess, but the `else if` clause (final one: lines 131-133) for SEGMENT is different than the other `defCutTypeValue`(s) ??

**Comment by scottkleinman** [comment]
That sounds like a logical explanation. I'll look into it later today.

**Comment by mleblanc321** [comment]
scott ... you are up so early :)

**Comment by scottkleinman** [comment]
I woke up hungry at 6:00. Annoying, but it gets my day started early.

**Comment by scottkleinman** [comment]
As a general note on this one, I kept all the original validation code from Lexos 2.5 that checks whether the user settings will generate more than 100 segments. So we might have to go through that systematically. As I recall, it's pretty dense code...

**Comment by scottkleinman** [comment]
I've had a little trouble reproducing your workflow. Here's what I have encountered:

1. Loading Daniel and Azarias together placed Daniel first in the file manager.
2. If I try to set the individual cutting options for Daniel to TOKENS=450 and the individual cutting options for Azarias to SEGMENTS=1, I get an error saying that I need to fill out the segment size in the main form.
3. If I set the main form to TOKENS=450, set the individual cutting options for Azarias to SEGMENTS=1, and ignore the individual cutting options for Daniel, I get the correct segmentation for Daniel and no errors.
4. If I set the main form to SEGMENTS=1, set the individual cutting options for Daniel to TOKENS=450, and ignore the individual cutting options for Azarias, I get the warning, but the correct segmentation still takes place.
5. If I set the main form to either TOKENS=450 or SEGMENTS=1, and do the same for the individual options, I get the warning, but the correct segmentation still takes place.

I think that summary matches and expands upon your experience. I'm hoping it helps with de-bugging.
